crypto/store/store_lib.c

```diff
@@ -146,6 +185,24 @@ OSSL_STORE_INFO *OSSL_STORE_load(OSSL_STORE_CTX *ctx)
             goto again;
     }
 
+    if (v != NULL && ctx->expected_type != 0) {
+        int returned_type = OSSL_STORE_INFO_get_type(v);
+
+        if (returned_type != OSSL_STORE_INFO_NAME && returned_type != 0) {
+            /*
+             * Soft assert here so those who want to harsly weed out faulty
+             * loaders can do so using a debugging version of libcrypto.
+             */
+            if (ctx->loader->expect != NULL)
+                assert(ctx->expected_type == returned_type);
```

Comment:
Is `ossl_assert()` appropriate here?

Reply:
Not sure what that'll give us in this particular case.

Reply:
Hm. Yeah - looking at it again, perhaps you're right. A NULL return here, just means there's nothing more to load right? i.e. not an actual error as such